Delegate filter query by empty value to filters (#6402)
sonata-project/SonataAdminBundle: UPGRADE-3.x.md

`ModelManagerInterface::getDefaultSortValues()` won't be used anymore.

### Empty values in datagrid filters

Now empty values are passed to datagrid filters. If you have custom datagrid filters, add empty string check to them.

> 

Author: @VincentLanglet please review

Reviewer: Adding an exemple could be helpful:

Something like

In the following exemple
```
->add('with_open_comments', CallbackFilter::class, [
    'callback' => static function(ProxyQueryInterface $queryBuilder, string $alias, string $field, array $value): bool {
        if (!$value['value']) {
            return false;
        }

        $queryBuilder
            ->leftJoin(sprintf('%s.comments', $alias), 'c')
            ->andWhere('c.status = :status')
            ->setParameter('status', Comment::STATUS_MODERATE);

         return true;
    },
    'field_type' => CheckboxType::class
 ]);
```
The `!$value['value']` check is required to not filtering by `''` if you didn't used the filter.

Author: I add this

`ModelManagerInterface::getDefaultSortValues()` won't be used anymore.

### Empty values in datagrid filters

> Now empty values are passed to datagrid filters. If you have custom datagrid filters, add empty string check to them.

Reviewer 2: ```suggestion
Empty values are passed to datagrid filters. If you have custom datagrid filters, you MUST add empty string checks to them.
```

```php
->add('with_open_comments', CallbackFilter::class, [

>     'callback' => static function(ProxyQueryInterface $queryBuilder, string $alias, string $field, array $value): bool {

Reviewer 2: ```suggestion
    'callback' => static function (ProxyQueryInterface $queryBuilder, string $alias, string $field, array $value): bool {
```

        if (!$value['value']) {
            return false;
        }

        $queryBuilder
            ->leftJoin(sprintf('%s.comments', $alias), 'c')
            ->andWhere('c.moderation = :moderation')
            ->setParameter('moderation', CommentModeration::APPROVED);

        return true;
    },

>     'field_type' => CheckboxType::class

Reviewer 2: ```suggestion
    'field_type' => CheckboxType::class,
```

]);
```

> The `!$value['value']` check is required to not filtering by `''` if you didn't used the filter.

Reviewer 2: ```suggestion
The `!$value['value']` check is required to avoid the filtering by `''` if you didn't used the filter.
```
